### PR TITLE
Update solc-0.4.23

### PR DESCRIPTION
**Pull request by orenyodfat**
solc  - 0.4.23
solium -  1.1.7
truffle  - 4.1.8
openzeppline-solidity - 1.9.0

Bump arc version to   0.0.0-alpha.39

Make sure solidity code lines does not exceed 145 characters.

### CODE REVIEW

File: contracts/VotingMachines/GenesisProtocol.sol [hunk 1]
```diff
@@ -109,7 +116,10 @@ contract GenesisProtocol is IntVoteInterface,UniversalScheme {
      * @param _proposer address
      * @return proposal's id.
      */
-    function propose(uint _numOfChoices, bytes32 , address _avatar, ExecutableInterface _executable,address _proposer) external returns(bytes32) {
+    function propose(uint _numOfChoices, bytes32 , address _avatar, ExecutableInterface _executable,address _proposer)
+        external
+        returns(bytes32)
+        {
```
Comment (leviadam) [inline contracts/VotingMachines/GenesisProtocol.sol]: Not according to solidity style guide

File: contracts/VotingMachines/AbsoluteVote.sol [hunk 2]
```diff
@@ -70,7 +70,10 @@ contract AbsoluteVote is IntVoteInterface {
      * @param _executable This contract will be executed when vote is over.
      * @return proposal's id.
      */
-    function propose(uint _numOfChoices, bytes32 _paramsHash, address _avatar, ExecutableInterface _executable,address) external returns(bytes32) {
+    function propose(uint _numOfChoices, bytes32 _paramsHash, address _avatar, ExecutableInterface _executable,address)
+        external
+        returns(bytes32)
+        {
```
Comment (leviadam) [inline contracts/VotingMachines/AbsoluteVote.sol]: Not according to solidity style guide

File: contracts/VotingMachines/AbsoluteVote.sol [hunk 3]
```diff
@@ -122,7 +125,12 @@ contract AbsoluteVote is IntVoteInterface {
    * @return bool true - the proposal has been executed
    *              false - otherwise.
    */
-    function ownerVote(bytes32 _proposalId, uint _vote, address _voter) external onlyProposalOwner(_proposalId) votable(_proposalId) returns(bool) {
+    function ownerVote(bytes32 _proposalId, uint _vote, address _voter)
+        external
+        onlyProposalOwner(_proposalId)
+        votable(_proposalId)
+        returns(bool)
+        {
```
Comment (leviadam) [inline contracts/VotingMachines/AbsoluteVote.sol]: Not according to solidity style guide

Comment (leviadam) [inline contracts/VotingMachines/AbsoluteVote.sol]: Not according to solidity style guide

File: contracts/VotingMachines/GenesisProtocol.sol [hunk 4]
```diff
@@ -86,7 +93,7 @@ contract GenesisProtocol is IntVoteInterface,UniversalScheme {
     /**
      * @dev Constructor
      */
-    function GenesisProtocol(StandardToken _stakingToken)
+    constructor(StandardToken _stakingToken)
     public
```
Comment (leviadam) [inline contracts/VotingMachines/GenesisProtocol.sol]: Can move to previous line

File: contracts/VotingMachines/GenesisProtocol.sol (hunk 1 above)
Comment (leviadam) [inline contracts/VotingMachines/GenesisProtocol.sol]: Not according to solidity style guide

File: contracts/controller/Controller.sol [hunk 5]
```diff
@@ -68,7 +68,7 @@ contract Controller is ControllerInterface {
     event AddGlobalConstraint(address indexed _globalConstraint, bytes32 _params,GlobalConstraintInterface.CallPhase _when);
     event RemoveGlobalConstraint(address indexed _globalConstraint ,uint256 _index,bool _isPre);
 
-    function Controller(
+    constructor(
         Avatar _avatar
```
Comment (leviadam) [inline contracts/controller/Controller.sol]: Can move to previous line

File: contracts/controller/UController.sol [hunk 6]
```diff
@@ -76,10 +76,13 @@ contract UController is ControllerInterface {
     event ExternalTokenIncreaseApproval (address indexed _sender, StandardToken indexed _externalToken, address _spender, uint _value);
     event ExternalTokenDecreaseApproval (address indexed _sender, StandardToken indexed _externalToken, address _spender, uint _value);
     event UpgradeController(address indexed _oldController,address _newController,address _avatar);
-    event AddGlobalConstraint(address indexed _globalConstraint, bytes32 _params,GlobalConstraintInterface.CallPhase _when,address indexed _avatar);
+    event AddGlobalConstraint(address indexed _globalConstraint,
```
Comment (leviadam) [inline contracts/controller/UController.sol]: Not according to solidity style guide

File: contracts/universalSchemes/ContributionReward.sol [hunk 7]
```diff
@@ -130,7 +125,12 @@ contract ContributionReward is UniversalScheme {
             _avatar.nativeToken().transferFrom(msg.sender, _avatar, controllerParams.orgNativeTokenFee);
         }
 
-        bytes32 contributionId = controllerParams.intVote.propose(2, controllerParams.voteApproveParams, _avatar, ExecutableInterface(this),msg.sender);
+        bytes32 contributionId = controllerParams.intVote.propose(2,
```
Comment (leviadam) [inline contracts/universalSchemes/ContributionReward.sol]: Not according to solidity style guide

File: contracts/universalSchemes/SchemeRegistrar.sol [hunk 8]
```diff
@@ -130,7 +126,12 @@ contract SchemeRegistrar is UniversalScheme {
         require(_scheme != address(0));
         Parameters memory controllerParams = parameters[getParametersFromController(_avatar)];
 
-        bytes32 proposalId = controllerParams.intVote.propose(2, controllerParams.voteRegisterParams, _avatar, ExecutableInterface(this),msg.sender);
+        bytes32 proposalId = controllerParams.intVote.propose(2,
```
Comment (leviadam) [inline contracts/universalSchemes/SchemeRegistrar.sol]: Not according to solidity style guide

**Comment by orenyodfat** [comment]
Thanks @leviadam .
Fixed style comments at 543c680

**Pull request merged**